src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -113,6 +114,16 @@ def _get_subparser(self, path):
                 self.subparsers[tuple(path[0:length])] = parent_subparser
         return parent_subparser
 
+    def _handle_command_package_error(self, err_msg): #pylint: disable=no-self-use
+        if err_msg and err_msg.startswith('argument _command_package: invalid choice:'):
+            import re
+            try:
+                possible_module = re.search("argument _command_package: invalid choice: '(.+?)'",
+                                            err_msg).group(1)
+                handle_module_not_installed(possible_module)
```

Comment:
Do we only invoke module discovery for top level commands? say `az foo`, or we do it at deeper level across, say `az resource group foo-bar`.
I hope we only do it for top level unless the discovery takes trivial time.